crates/sage/src/scoring.rs

```diff
@@ -53,6 +55,8 @@ impl AddAssign<InitialHits> for InitialHits {
 /// Features of a candidate peptide spectrum match
 pub struct Feature {
     #[serde(skip_serializing)]
+    // psm_id help to match with matched fragments table.
+    pub psm_id: Option<usize>,
```

Comment:
I think we should just make this a required field - then it will always be written to output files, and should simplify building parquet schemas, etc